docs/modules/ROOT/pages/type-definitions/indexes-and-constraints.adoc

```diff
@@ -59,10 +59,85 @@ type Colour @node(label: "Color", additionalLabels: ["Hue"]) {
 }
 ----
 
-[[type-definitions-constraints-asserting]]
+[[type-definitions-indexes-fulltext]]
+== Fulltext indexes
+
+You can use the `@fulltext` directive to add a https://neo4j.com/docs/cypher-manual/current/indexes-for-full-text-search/[Full text index] inside Neo4j. 
+
+[source, graphql, indent=0]
+----
+input FullTextInput {
+  name: String!
+  fields: [String]!
+  defaultThreshold: Int
+}
+
+"""
+Informs @neo4j/graphql that there should be a fulltext index in the database, allows users to search by the index in the generated schema.
+"""
+directive @fulltext(indexes: [FullTextInput]!) on OBJECT
+----
+
+Using this directive does not automatically ensure the existence of these indexes, and you will need to run a function on server startup. See the section xref::type-definitions/indexes-and-constraints.adoc#type-definitions-indexes-and-constraints-asserting[Asserting constraints] below for details.
+
+=== Specifying the `@fulltext` directive 
+
+The directive can be used on nodes. Here we add a Fulltext index, called 'ProductName', for the name field, on the Product node.
+
+[source, graphql, indent=0]
+----
+type Product @fulltext(indexes: [{ name: "ProductName", fields: ["name"] }]) {
+    name: String!
+    color: Color @relationship(type: "OF_COLOR", direction: OUT)
+}
+----
+
+When you run xref::type-definitions/indexes-and-constraints.adoc#type-definitions-indexes-and-constraints-asserting[Asserting constraints] this shall do the index creation like so: 
+
+[source, cypher, indent=0]
+----
+CALL db.index.fulltext.createNodeIndex("ProductName", ["Product"], ["name"])
+----
+
+=== Using the `@fulltext` directive 
+
+Once you specify the index, you will now gain a 'Top Level' `search` key on the following operations: 
+
+1. read
+2. update
+3. delete
+4. count
+5. aggregate
+
+Here we use the `search` key, and the phrase is using https://lucene.apache.org/[Lucene’s full-text query language] to match and return Products: 
```

Comment:
Where did we land on the `fulltext` vs `search` naming?

Reply:
We haven't what is peoples thoughts @oskarhane @angrykoala  ?

Reply:
Personally, I think search is too broad, as it can refer to other kinds of search, not only fulltext